### PR TITLE
VPLAY-11984: multi-period video-only test stream crashing

### PR DESCRIPTION
**Pull request by nu641001**
Reason for change: Updating profileIdxForBandwidthNotification after period switch to avoid heap-buffer-overflow in multi-period video-only test stream. Risks: Low
Test Procedure: Test with multi-period video-only test stream in VPLAY-11984 Priority: P1
Signed-off-by: Nandakishor U M <nu641001@gmail.com>

### CODE REVIEW

**Review by Vinish100 — COMMENTED** [review]
Looks good to me 👍

File: fragmentcollector_mpd.cpp [hunk 1]
```diff
@@ -7136,14 +7136,14 @@ AAMPStatusType StreamAbstractionAAMP_MPD::UpdateTrackInfo(bool modifyDefaultBW,
 				{
 					mUpdateStreamInfo = false;
 					vector<Representation *> representations = mMPDParseHelper->GetBitrateInfoFromCustomMpd(pMediaStreamContext->adaptationSet);
-					int representationCount = (int)representations.size();
-					if ((representationCount != mBitrateIndexVector.size()) && mStreamInfo)
+					size_t representationCount = representations.size();
```
[the remark this answers is not in the document]
Comment (nu641001) [reply fragmentcollector_mpd.cpp:7142]: Done

File: test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp [hunk 2]
```diff
@@ -2194,6 +2214,163 @@ TEST_F(StreamAbstractionAAMP_MPDTest, GetStreamInfoTest) {
...
+
+	// Now: Switch to source content with only 1 profile (heap size reduction)
+	mStreamAbstractionAAMP_MPD->ResizeStreamInfo(1);
+	mStreamAbstractionAAMP_MPD->GetStreamInfoAt(0).bandwidthBitsPerSecond = 2000000;
+
+	// Update ABR manager to reflect new profile count
+	EXPECT_CALL(*g_mockABRManager, getProfileCount())
+		.WillRepeatedly(Return(1));
+
+	// Test: Old cached profileIndex=4 should now be safely handled
```
[the remark this answers is not in the document]
Comment (nu641001) [reply test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp:2354]: This is to test that even if ABR is corrupted, the API shoudn't fail.
Already expectations are set on top

File: test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp [hunk 3]
```diff
@@ -2194,6 +2214,163 @@ TEST_F(StreamAbstractionAAMP_MPDTest, GetStreamInfoTest) {
...
+	StreamInfo *result = mStreamAbstractionAAMP_MPD->CallGetStreamInfo(4);
+	EXPECT_EQ(result, nullptr); // Should return nullptr due to invalid user data
+
+	// Valid profile should still work
+	StreamInfo *validResult = mStreamAbstractionAAMP_MPD->CallGetStreamInfo(0); // Profile 0 exists
+	ASSERT_NE(validResult, nullptr);
+	EXPECT_EQ(validResult->bandwidthBitsPerSecond, 2000000);
+
+	// Test additional edge cases
```
[the remark this answers is not in the document]
Comment (nu641001) [reply test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp:2363]: // Verify all profiles are accessible through ABR manager
	for (int i = 0; i < 5; i++)
	{
		StreamInfo *result = mStreamAbstractionAAMP_MPD->CallGetStreamInfo(i);
		ASSERT_NE(result, nullptr);
	}
	This sets expectations, and a test is created to check the API even if ABR is corrupted

File: fragmentcollector_mpd.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (nu641001) [reply fragmentcollector_mpd.cpp:7142]: Done

File: test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (nu641001) [reply test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp:2354]: // Verify all profiles are accessible through ABR manager
	for (int i = 0; i < 5; i++)
	{
		StreamInfo *result = mStreamAbstractionAAMP_MPD->CallGetStreamInfo(i);
		ASSERT_NE(result, nullptr);
	}
	This sets expectations, and a test is created to check the API even if ABR is corrupted

File: test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp [hunk 4]
```diff
@@ -2194,6 +2214,163 @@ TEST_F(StreamAbstractionAAMP_MPDTest, GetStreamInfoTest) {
...
+	// Mock ABR manager to simulate initial state with 5 profiles
+	EXPECT_CALL(*g_mockABRManager, getProfileCount())
+		.WillRepeatedly(Return(5));
+
+	// Set up user data mapping for initial 5 profiles
+	for (int i = 0; i < 5; i++)
+	{
+		EXPECT_CALL(*g_mockABRManager, getUserDataOfProfile(i))
+			.WillRepeatedly(Return(i)); // Maps to index i in mStreamInfo
+	}
```
[the remark this answers is not in the document]
Comment (nu641001) [reply test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp:2337]: getUserDataOfProfile has argument int, So keeping it

**Review by pstroffolino — APPROVED** [review]

File: test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp [hunk 5]
```diff
@@ -2194,6 +2214,163 @@ TEST_F(StreamAbstractionAAMP_MPDTest, GetStreamInfoTest) {
...
+
+	// Verify all profiles are accessible through ABR manager
+	for (int i = 0; i < 5; i++)
+	{
+		StreamInfo *result = mStreamAbstractionAAMP_MPD->CallGetStreamInfo(i);
+		ASSERT_NE(result, nullptr);
+	}
```
[the remark this answers is not in the document]
Comment (nu641001) [reply test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp:2344]: GetStreamInfo takes argument as integer. So keeping it

File: test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp [hunk 6]
```diff
@@ -2194,6 +2214,163 @@ TEST_F(StreamAbstractionAAMP_MPDTest, GetStreamInfoTest) {
...
+	// Set non-FOG TSB mode to enable ABR manager path
+	mStreamAbstractionAAMP_MPD->SetIsFogTSB(false);
+	mStreamAbstractionAAMP_MPD->SetAdPlayingFromCDN(false);
+
+	// First: Set up ad content with 5 profiles
+	mStreamAbstractionAAMP_MPD->ResizeStreamInfo(5);
+	for (int i = 0; i < 5; i++)
+	{
+		mStreamAbstractionAAMP_MPD->GetStreamInfoAt(i).bandwidthBitsPerSecond = (i + 1) * 1000000;
+	}
```
[the remark this answers is not in the document]
Comment (nu641001) [reply test/utests/tests/StreamAbstractionAAMP_MPD/FunctionalTests.cpp:2326]: Casted to size_t on call

File: fragmentcollector_mpd.cpp [hunk 7]
```diff
@@ -10159,7 +10163,7 @@ StreamAbstractionAAMP_MPD::~StreamAbstractionAAMP_MPD()
 
 	aamp->SyncBegin();
 
-	SAFE_DELETE_ARRAY(mStreamInfo);
+	// mStreamInfo is now a vector and will be automatically destroyed
```
[the remark this answers is not in the document]
Comment (nu641001) [reply fragmentcollector_mpd.cpp:10169]: Comment not needed. Hence, can be Removed
